src/test_workflow/integ_test/integ_test_suite_opensearch_dashboards.py

```diff
@@ -81,6 +94,38 @@ def __setup_cluster_and_execute_test_config(self, config: str) -> int:
             os.chdir(self.work_dir)
             return self.execute_integtest_sh(endpoint, port, security, config)
 
+    def multi_execute_integtest_sh(self, cluster_endpoints: list, security: bool, test_config: str) -> int:
+        script = ScriptFinder.find_integ_test_script(self.component.name, self.repo.working_directory)
+
+        def custom_node_endpoint_encoder(node_endpoint: NodeEndpoint) -> dict:
+            return {"endpoint": node_endpoint.endpoint, "port": node_endpoint.port, "transport": node_endpoint.transport}
+        if os.path.exists(script):
+            single_data_node = cluster_endpoints[0].data_nodes[0]
+            cmd = f"bash {script} -b {single_data_node.endpoint} -p {single_data_node.port} -s {str(security).lower()} -t {self.component.name} -v {self.bundle_manifest.build.version}"
+            self.repo_work_dir = os.path.join(
+                self.repo.dir, self.test_config.working_directory) if self.test_config.working_directory is not None else self.repo.dir
+            (status, stdout, stderr) = execute(cmd, self.repo_work_dir, True, False)
+            test_result_data = TestResultData(
+                self.component.name,
+                test_config,
+                status,
+                stdout,
+                stderr,
+                self.test_artifact_files
+            )
+            self.save_logs.save_test_result_data(test_result_data)
+            if stderr:
+                logging.info("Stderr reported for component: " + self.component.name)
+                logging.info(stderr)
+            return status
+        else:
+            logging.info(f"{script} does not exist. Skipping integ tests for {self.component.name}")
+            return 0
```

Comment:
Nit: I kinda feel like this entire part can be put into another wrapper to be called.

Reply:
Do you think it's better to have a different function to implement this part @peterzhuamazon ?
If that's the case we can implement this in the base class both for OS and OSD.

Reply:
I think this is okay for now (since it's a nit). We can revisit this later.

Reply:
> Do you think it's better to have a different function to implement this part @peterzhuamazon ? If that's the case we can implement this in the base class both for OS and OSD.

That is what I mean as a improvement later.